src/commands/interactive/untagged-tenants.ts

```diff
@@ -0,0 +1,311 @@
+        console.log(help);
+        break;
+      }
+      case "back": {
+        running = false;
+        console.clear();
+        break;
+      }
+      case "quit": {
+        Deno.exit();
+        break;
+      }
+      default: {
+        throw new MeshError("Invalid value. Something went horribly wrong.");
+      }
+    }
+  }
+}
+
+async function getData(
+  options: CmdGlobalOptions,
+  from: string | undefined = undefined,
+  to: string | undefined = undefined,
+) {
+  setupLogger(options);
+  await verifyCliAvailability();
+
+  if ((from == undefined || "") && (to == undefined || "")) {
+    const config = loadConfig();
+    const meshAdapterFactory = new MeshAdapterFactory(config);
+    const queryStatistics = new QueryStatistics();
+    const meshAdapter = meshAdapterFactory.buildMeshAdapter(
+      options,
+      queryStatistics,
+    );
+
+    const allTenants = await meshAdapter.getMeshTenants();
+    return allTenants;
+  } else if (from != undefined) {
+    const config = loadConfig();
+    const meshAdapterFactory = new MeshAdapterFactory(config);
+    const queryStatistics = new QueryStatistics();
+    const meshAdapter = meshAdapterFactory.buildMeshAdapter(
+      options,
+      queryStatistics,
+    );
+
+    // We create UTC dates because we do not work with time, hence we do not care about timezones.
+    const start = moment.utc(from).startOf("day").toDate();
+    if (isNaN(start.valueOf())) {
+      throw new MeshError(
+        `You have entered an invalid date for '--from':  ${from}`,
+      );
+    }
+    const end = moment.utc(to).endOf("day").toDate();
+    if (isNaN(start.valueOf())) {
+      throw new MeshError(
+        `You have entered an invalid date for '--to': ${to}`,
+      );
+    }
+
+    // Every of these methods can throw e.g. because a CLI tool was not installed we should think about
+    // how to do error management to improve UX.
+    const allTenants = await meshAdapter.getMeshTenants();
+
+    await meshAdapter.attachTenantCosts(allTenants, start, end);
+    return allTenants;
+  } else {
+    throw new MeshError("Something went horribly wrong.");
+  }
+}
+
+async function showUntagged(
+  options: CmdGlobalOptions,
+  data: MeshTenant[],
+  from: string | undefined = undefined,
+  to: string | undefined = undefined,
+  sortByCost: boolean,
+) {
+  console.clear();
+  const tags = getAllTags(data);
+  let running = true;
+
+  if (sortByCost) {
+    data = sortTenantDataByCost(options, data);
+  } else {
+    data = sortTenantDataByName(options, data);
+  }
+  while (running) {
+    const selectedTag = await selectTag(tags);
+    if (selectedTag == "BACK") {
+      running = false;
+    } else {
+      const untaggedTenant: Array<MeshTenant> | undefined = [];
+
+      for (const tenant of data) {
+        if (filterForTag(options, tenant, selectedTag) == false) {
+          untaggedTenant.push(tenant);
+        }
+      }
+      let runningInside = true;
+      while (runningInside) {
```

Comment:
t: I suggest to use a slightly cleaner tag name. Even though the usage is very "close" and not scattered along, so its not super bad, but generally I like it a bit clearer. Something like `isRunningTenantSelection` (also the is* prefix is often used for boolean flags).

Reply:
see this